webapp/app/assessment/components/section/components/descriptions/components/useDescription.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import * as R from 'ramda'
+
+import { debounce } from '@webapp/utils/functionUtils'
+
+import useGetRequest from '@webapp/components/hooks/useGetRequest'
+import usePostRequest from '@webapp/components/hooks/usePostRequest'
+import useCountryIso from '@webapp/components/hooks/useCountryIso'
+import useOnUpdate from '@webapp/components/hooks/useOnUpdate'
+import * as autosave from '@webapp/app/components/autosave/actions'
+
+const getUrl = (countryIso, section, name) => `/api/country/descriptions/${countryIso}/${section}/${name}`
+const getData = name => R.pathOr(null, [name, 'content'])
+
+export default (name, section, template) => {
+  const dispatch = useDispatch()
+  const countryIso = useCountryIso()
+
+  // ====== data read
+  const { data = template || null, setState, loading, dispatch: fetchData } = useGetRequest(
+    getUrl(countryIso, section, name)
+  )
+  // ====== data update
+  const { dispatch: postData, loaded: postDataLoaded } = usePostRequest(getUrl(countryIso, section, name), {
+    content: getData(name)(data),
+  })
+
+  const onChange = content => {
+    dispatch(autosave.start)
+    setState({ data: { [name]: { content } } })
+    debounce(postData, 'postDescriptionData', 800)()
```

Comment:
add the name to debounce id. user might start editing another description while this one is still posting